index/scorch/segment/zap/contentcoder.go

```diff
@@ -94,9 +93,21 @@ func (c *chunkedContentCoder) flushContents() error {
 		return err
 	}
 
+	// convert the document data lens to data offsets
+	if len(c.chunkMeta) > 1 {
+		var runningOffset uint64
+		var index, i int
+		for i = 1; i < len(c.chunkMeta); i++ {
+			runningOffset += c.chunkMeta[i-1].DocDvOffset
+			c.chunkMeta[index].DocDvOffset = runningOffset
+			index++
+		}
+		c.chunkMeta[index].DocDvOffset = c.chunkMeta[i-1].DocDvOffset
```

Comment:
I'm a little confused about the serialized encoding -- perhaps some doc commentary would help?

I suspect the chunkMeta.DocDvOffset field is "switching gears" or switching meanings somewhere along the way?  I suspect it starts off as a length but is getting transformed?

Reply:
we are following the same offset slice pattern as that used with loadChunk here, Let me add some comments there.. 
Yes, you are right, the length slice is getting transformed to a starting offset slice with the length of the last item getting stored in last position.